src/services/services.ts

```diff
@@ -3504,7 +3504,11 @@ namespace ts {
                         // We don't want to complete using the type acquired by the shape
                         // of the binding pattern; we are only interested in types acquired
                         // through type declaration or inference.
-                        if (rootDeclaration.initializer || rootDeclaration.type) {
+                        // Also proceed if rootDeclaration is parameter and if its containing function expression\arrow function is contextually typed -
+                        // type of parameter will flow in from the contextual type of the function
+                        if (rootDeclaration.initializer ||
+                            rootDeclaration.type ||
+                            (rootDeclaration.kind === SyntaxKind.Parameter && isExpression(rootDeclaration.parent) && typeChecker.getContextualType(<Expression>rootDeclaration.parent))) {

```

Comment:
object literal properties are handled in the branch above.
The check if parent has contextual type is necessary to make sure that the type that we'll obtain from `getTypeAtLocation` is backed up by something (contextual type of the function) and not just implied by the shape of object binding pattern